servicetalk-http-api/src/main/java/io/servicetalk/http/api/HeaderUtils.java

```diff
@@ -318,7 +318,7 @@ public static boolean pathMatches(final CharSequence requestPath, @Nullable fina
         }
 
         if (requestPath.length() == cookiePath.length()) {
-            return requestPath.equals(cookiePath);
+            return contentEquals(requestPath, cookiePath);
```

Comment:
Good catch 👍

Reply:
hmm... should this compare ignoring case ?

Reply:
Good question, I looked it up and its not explicit. Also the previous solution was case sensitive too.

Some of the sources [1][2] treats the components as case sensitive.

1. https://tools.ietf.org/html/draft-ietf-httpbis-p1-messaging-25#section-2.7.3
>... The scheme and host are case-insensitive and normally provided in lowercase; all other components are compared in a case-sensitive manner. ...

2. https://tools.ietf.org/html/rfc3986#section-6.2.2.1
> ... The other generic syntax components are assumed to be case-sensitive unless specifically defined otherwise by the scheme ...